safekeeper/src/pull_timeline.rs

```diff
@@ -46,6 +47,7 @@ pub async fn stream_snapshot(
     source: NodeId,
     destination: NodeId,
     tx: mpsc::Sender<Result<Bytes>>,
+    storage: Arc<GenericRemoteStorage>,
```

Comment:
Here, we can also accept a WalBackup or optional remote storage. `stream_snapshot_resident_guts` can work just fine without remote storage, but if there is a partial segment it should error out if remote storage isn't configured.

For the `stream_snapshot_offloaded_guts` we can error out immediately if remote storage isn't configured, since offloaded timelines require remote storage.

Reply:
https://github.com/neondatabase/neon/pull/10179/commits/74429d400841e41c0202ab985a2ae2573be7dec2

Reply:
My understanding is that the above commit inside this PR also fixes
- https://github.com/neondatabase/neon/issues/11501